src/viewers/VolumeViewer.js

```diff
@@ -0,0 +1,101 @@
+import React, { useMemo } from 'react'; // eslint-disable-line import/no-unresolved
+import { Matrix4 } from 'math.gl';
+
+import { getPhysicalSizeScalingMatrix } from '../layers/utils';
+import VivViewer from './VivViewer';
+import { VolumeView } from '../views';
+import { RENDERING_MODES } from '../constants';
+
+/**
+ * This component provides a volumetric viewer that provides provides volume-ray-casting.
+ * @param {Object} props
+ * @param {Array} props.sliderValues List of [begin, end] values to control each channel's ramp function.
+ * @param {Array} props.colorValues List of [r, g, b] values for each channel.
+ * @param {Array} props.channelIsOn List of boolean values for each channel for whether or not it is visible.
+ * @param {string} [props.colormap] String indicating a colormap (default: '').  The full list of options is here: https://github.com/glslify/glsl-colormap#glsl-colormap
+ * @param {Array} props.loader This data source for the viewer. PixelSource[]. If loader.length > 1, data is assumed to be multiscale.
+ * @param {Array} props.loaderSelection Selection to be used for fetching data
+ * @param {Array} [props.resolution] Resolution at which you would like to see the volume and load it into memory (0 highest, loader.length - 1 the lowest with default loader.length - 1)
+ * @param {import('./VivViewer').ViewStateChange} [props.onViewStateChange] Callback that returns the deck.gl view state (https://deck.gl/docs/api-reference/core/deck#onviewstatechange).
+ * @param {Array} [props.renderingMode] One of Maximum Intensity Projection, Minimum Intensity Projection, or Additive
+ * @param {Matrix4} [props.modelMatrix] A column major affine transformation to be applied to the volume.
+ * @param {Array} [props.xSlice] 0-1 interval on which to slice the volume.
+ * @param {Array} [props.ySlice] 0-1 interval on which to slice the volume.
+ * @param {Array} [props.zSlice] 0-1 interval on which to slice the volume.
```

Comment:
I think the reason was that scaling matrices would cause these two concepts to not necessarily align (i.e if you inflate the z-direction to match the physical size).